### DISCUSSION TITLE
Error object has no attibute "prettify "

### DISCUSSION THREAD

**Discussion by moogli90** [teamalgoritma/community#198] [Q&A] [answered]
bagaimana mengatasi error ini?
![image](https://user-images.githubusercontent.com/107478291/178170833-b7721630-1ee3-4986-9229-b4aa6873a415.png)

**Comment by triarm** [comment, marked as answer]
Hi @moogli90 ,
error tersebut terjadi karena key yang digunakan untuk melakukan scraping tidak terdapat pada elemen HTML di website yang akan dilakukan scraping. Silahkan untuk dicoba key yang lain sehingga diperoleh data yang Anda inginkan.

Demikian yang dapat kami sampaikan. Semoga menjawab.

**Reply by moogli90** [reply]
di petunjuk & video capstone hanya memakai preetify, apakah ada fungsi lain? mohon masukannya

**Reply by triarm** [reply]
Untuk bagian prettifynya sudah tepat, namun yang perlu diganti adalah bagian keynya, yaitu `"div"` dan `attrs = {"table":"card-block"}`. Silahkan untuk mengganti key tersebut sesuai dengan elemen yang ada pada website yang akan dilakukan scrapping.

Semoga menjawab

**Reply by moogli90** [reply]
sy sudah coba ganti ternyata sama saja, bgmn ya
div, th, table dll

**Reply by moogli90** [reply]
semua masih error

**Reply by triarm** [reply]
![image](https://user-images.githubusercontent.com/20942472/178183694-1c917940-2df1-452a-a76d-87695df0a86c.png)

Jika dilihat pada struktur berikut, data disimpan pada tag `table` dan class `table table-striped text-sm text-lg-normal`. Bapak dapat mencoba memanfaatkan informasi tersebut sebagai key yang digunakan untuk melakukan scraping. Semoga membantu :)

**Reply by triarm** [reply]
Jika masih terdapat error, Bapak dapat mengirimkan file Jupyter Notebook (.ipynb) agar dapat kami bantu lebih lanjut terkait error yang dialami.

**Reply by moogli90** [reply]
itu td jg sebelumnya jg sdh sy coba namun msh error
![image](https://user-images.githubusercontent.com/107478291/178186394-d1e715c8-5a8b-4fd6-a284-f125f090c55c.png)

**Reply by triarm** [reply]
Apakah Bapak dapat melampirkan screenshot cell bagian request URLnya? (cell yang terdapat code `requests.get(<link>)`)?

**Reply by moogli90** [reply]
ini kak
![image](https://user-images.githubusercontent.com/107478291/178186897-1b400705-cdd8-4a98-b616-d2420569f02e.png)

**Reply by triarm** [reply]
Pada bagian request data, Bapak dapat menambahkan code berikut agar dapat mengakses data pada website tersebut
```{python}
url_get = requests.get('https://www.coingecko.com/en/coins/ethereum/historical_data#panel',  headers = { 'User-Agent': 'Popular browser\'s user-agent', })
```
Setelah itu silahkan dicoba kembali code selanjutnya

**Reply by moogli90** [reply]
baik kak, kenapa ahrus ditambahkan itu ya? dan kenapa di ptunjuknya tidak ada?

**Reply by triarm** [reply]
Terkait penambahan code tersebut karena saat ini website coingecko menambah proteksi baru terhadap websitenya sehingga proses requests tidak dapat dilakukan ketika melakukan requests biasa
